crates/loam-cli/examples/soroban/nft/src/nft.rs

```diff
@@ -0,0 +1,118 @@
+use loam_sdk::{
+    soroban_sdk::{self, contracttype, env, Address, Bytes, Lazy, Map, Vec},
+    IntoKey,
+};
+
+use crate::subcontract::{IsInitable, IsNonFungible};
+
+#[contracttype]
+#[derive(IntoKey)]
+pub struct MyNonFungibleToken {
+    admin: Address,
+    name: Bytes,
+    total_count: u128,
+    owners_to_nft_ids: Map<Address, Map<u128, ()>>, // the owner's collection
+    nft_ids_to_owners: Map<u128, Address>,
+    nft_ids_to_metadata: Map<u128, Bytes>,
+}
+
+impl MyNonFungibleToken {
+    #[must_use]
+    pub fn new(admin: Address, name: Bytes) -> Self {
+        MyNonFungibleToken {
+            admin,
+            name,
+            total_count: 0,
+            owners_to_nft_ids: Map::new(env()),
+            nft_ids_to_owners: Map::new(env()),
+            nft_ids_to_metadata: Map::new(env()),
+        }
+    }
+}
+
+impl Default for MyNonFungibleToken {
+    fn default() -> Self {
+        MyNonFungibleToken::new(env().current_contract_address(), Bytes::new(env()))
+    }
+}
+
+impl IsInitable for MyNonFungibleToken {
+    fn nft_init(&self, admin: Address, name: Bytes) {
+        self.admin.require_auth();
```

Comment:
Wait, how is `self.admin` already set? And why allow `admin` to be passed in here, if it's already set? Should `admin` even be part of the `nft_init` params?

Reply:
Having two different kinds of admin—"the one that updates the contract source" vs "the one that mints tokens"—doesn't seem useful?

Reply:
@elizabethengelman here's how @willemneal helped me implement this in EquitX, for now:

- no `admin` attribute on any subcontract other than the `Core` / `Admin` one
- add a [`require_auth` function](https://github.com/EquitXDev/equitxdev/blob/3ac25cd079a0a9831ab6ed12cd94b53d19d417ac/contracts/data-feed/src/lib.rs#L19-L25) to `Contract` 
- while implementing other subcontracts, I can then `use crate::Contract` and [call `Contract::require_auth()`](https://github.com/EquitXDev/equitxdev/blob/3ac25cd079a0a9831ab6ed12cd94b53d19d417ac/contracts/data-feed/src/data_feed.rs#L78)

While the dependencies are sort of circular, this allows one subcontract to make calls into another.

Reply:
awesome, thanks so much!